### PR TITLE
Bug 1286859 - Add a 'cancel' link for spark jobs

### PR DESCRIPTION
**Pull request by washort**
See also https://github.com/mozilla/emr-bootstrap-spark/pull/110, necessary for this to work

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/jupyter-spark/26)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: src/jupyter_spark/static/extension.js [hunk 1]
```diff
@@ -219,18 +219,23 @@ define([
...
         var progress_count = cell.element.find('.progress_counter');
         if (progress_count.length < 1) {
             console.log("No progress counter found");
         };
         var job_name = "";
+        var canceller = null;
         if (spark_is_running) {
             cell_jobs_counter = cache[0].jobs.length - jobs_in_cache;
             job_name =  ": " + cache[0].jobs[0].name
+            canceller = $('<a href="#"> Cancel</a>').addClass('spark_canceller').on(
```
Comment (robhudson) [inline src/jupyter_spark/static/extension.js]: Is the extra space inside the link tag intentional?

Comment (robhudson) [inline src/jupyter_spark/static/extension.js]: I don't see the class used at all. Is that required?

**Comment by robhudson** [comment]
@washort think we can get the tests to pass on this project and rebase?

**Comment by washort** [comment]
Looks like tox-travis fixed it. r?

**Comment by florian** [comment]
Can this be merged? It'd be super useful and it looks like the tests now pass.

**Comment by mreid-moz** [comment]
@mdboom maybe this is something you could have a look at?

**Comment by mdboom** [comment]
Just had a look at this.  I can definitely confirm it still works.
A couple of thoughts:

1. Minor point: I think it would be good to add a little sprinkle of CSS so the cancel link looks more "clickable".

2. It's a shame that canceling spews a big scary Javaesque traceback at you, but looking at how this all works, I doubt it's easy to fix that.  The kernel doesn't really know how or why the task was canceled, just that it was.  I think we shouldn't let that hold up this PR.  Maybe create a new "papercut" issue after merging this?

@washort: I'm happy to do (1) to push this over the finish line, but don't want to step on toes...

**Comment by washort** [comment]
Go ahead, I haven't been involved in this side of things for a while. Thanks!

**Comment by mdboom** [comment]
Closing.  Replacing with #34.

**Pull request closed without merging**